core/eval.go

```diff
@@ -77,6 +79,9 @@ func evalSET(args []string) []byte {
 	for i := 2; i < len(args); i++ {
 		switch args[i] {
 		case "EX", "ex":
+			if exDurationMs != -1 {
```

Comment:
I feel checking with `-1` is sufficient and understandable unless there is some edge case where this would break.
But it's your call since I do not have much "production" experience in Go.

PS: Thanks for introducing me to the concept of `iota` in Go.

Reply:
> I feel checking with -1 is sufficient and understandable unless there is some edge case where this would break.

Couple of reasons I feel the above approach is better:
1. A user can send -1 as a value, so -1 doesn't really represent an `uninitialized` value.
2. As I mentioned earlier the code becomes more readable